spring/src/main/java/cucumber/runtime/java/spring/SpringFactory.java

```diff
@@ -73,6 +87,20 @@ public boolean addClass(final Class<?> stepClass) {
         return true;
     }
 
+    private void checkNoComponentAnnotations(Class<?> type) {
```

Comment:
This is not going to work. This should recursively search the annotations annotations for `@Component` taking care not to get stuck in a loop.